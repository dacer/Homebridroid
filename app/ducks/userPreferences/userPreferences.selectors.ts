import { pathOr } from "ramda";

import { namespace } from "./userPreferences.slice";

const getServerAddress = (state: any) =>
pathOr("", [namespace, "serverAddress"], state);

export default {
  getServerAddress
};
