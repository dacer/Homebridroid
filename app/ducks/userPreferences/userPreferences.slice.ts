import { createSlice, PayloadAction } from "@reduxjs/toolkit";


const initialState = {
  username: "",
  password: "",
  serverAddress: "",
};

const userPreferences = createSlice({
  name: "userPreferences",
  initialState,
  reducers: {
    userLoggedIn(state, action: PayloadAction<AuthInfo>) {
      state.username = action.payload.username
      state.password = action.payload.password
      state.serverAddress = action.payload.serverAddress
    },
  },
});

const {
  userLoggedIn,
} = userPreferences.actions;

export const actions = {
  userLoggedIn,
};

export const namespace = userPreferences.name;

export const reducer = userPreferences.reducer;

export interface AuthInfo {
  username: string,
  password: string,
  serverAddress: string,
  token?: string
}