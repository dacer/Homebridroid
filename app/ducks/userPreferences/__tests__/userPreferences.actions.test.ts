import userPreferences from "../";

describe("userPreferences actions should", () => {
  it("export expected actions", () =>
    expect(userPreferences.actions).toMatchSnapshot());

  it("be able to update to a new term of use version", () => {
    const username = "username";
    const password = "password";
    const serverAddress = "http://192.168.1.111";

    const expectedAction = {
      type: userPreferences.actions.userLoggedIn.toString(),
      payload: {username, password, serverAddress},
    };
    expect(
      userPreferences.actions.userLoggedIn({username, password, serverAddress})
    ).toEqual(expectedAction);
  });

});
