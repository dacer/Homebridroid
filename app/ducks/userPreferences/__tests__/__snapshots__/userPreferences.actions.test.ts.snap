// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userPreferences actions should export expected actions 1`] = `
Object {
  "userLoggedIn": [Function],
}
`;
